net/radius-mac/files/radius-mac.init

```diff
@@ -0,0 +1,207 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2025 Brian 'redbeard' Harrington <redbeard@dead-city.org>
+# This is free software, licensed under the MIT License
+
+START=99
+USE_PROCD=1
+
+PROG=/usr/bin/radius-mac
+CONFIG_BASE_DIR="/etc" # Directory for generated .ini files
+
+# Helper to determine the path for the generated .ini file
+# $1: section_name (from UCI, e.g., 'default', 'guest_server')
+# $2: variable name to store the result (output parameter)
+get_cfg_file_path() {
+    local section_name="$1"
+    local __result_var="$2"
+    local cfg_path
+
+    if [ "$section_name" = "default" ]; then
+        cfg_path="$CONFIG_BASE_DIR/radius-mac.ini"
+    else
+        cfg_path="$CONFIG_BASE_DIR/radius-mac-$section_name.ini"
+    fi
+    eval "$__result_var='$cfg_path'"
+}
+
+# Validation helper functions
+_log_error() {
+    logger -t radius-mac-init "Error: $1"
+}
+
+_is_valid_number_in_range() {
+    local val="$1"
+    local min="$2"
+    local max="$3"
+    local context="$4" # For logging
+
+    if ! echo "$val" | grep -qE '^[0-9]+$'; then
+        _log_error "$context: Value '$val' is not a number."
+        return 1
+    fi
+    if [ "$val" -lt "$min" ] || [ "$val" -gt "$max" ]; then
+        _log_error "$context: Value '$val' is out of range ($min-$max)."
+        return 1
+    fi
+    return 0
+}
+
+_is_valid_string_length() {
+    local val="$1"
+    local min_len="$2"
+    local max_len="$3"
+    local context="$4" # For logging
+    local len=${#val}
+
+    if [ "$len" -lt "$min_len" ] || [ "$len" -gt "$max_len" ]; then
+        _log_error "$context: Length '$len' is out of range ($min_len-$max_len)."
+        return 1
+    fi
+    return 0
+}
+
+_is_valid_ipv4() {
```

Comment:
I think I might not have a good question, but IPv6 is not currently supported in this package?

Reply:
Discussion and a reference to the potential implementation will be tracked here: https://github.com/carlanton/radius-mac/issues/5